Bottom: 2ba7580c0edd1ff25bf0aa2d86b5f3726bd3c579
Top:    dccf695dfe5e751cd6327fd3cc3cd126bd8538b7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-13 22:26:47 +0000

Refresh of polygon-intersection-additions

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 6455bdf..671b7f8 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1058,9 +1058,6 @@ vertex_in_seg_rounded (const BoxType * b, void *cl)
   struct seg *s = (struct seg *) b;
   VNODE *new_node;
 
-  if (!line_point_inters (EDGE_BACKWARD_VERTEX (s->v)->point, EDGE_FORWARD_VERTEX (s->v)->point, i->v->point))
-    return 0;
-
   /* When new nodes are added at the end of a pass due to an intersection
    * the segments may be altered. If either segment we're looking at has
    * already been intersected this pass, skip it until the next pass.
@@ -1068,6 +1065,9 @@ vertex_in_seg_rounded (const BoxType * b, void *cl)
   if (s->intersected)
     return 0;
 
+  if (!line_point_inters (EDGE_BACKWARD_VERTEX (s->v)->point, EDGE_FORWARD_VERTEX (s->v)->point, i->v->point))
+    return 0;
+
   if (i->touch)			/* if checking touches one find and we're done */
     longjmp (*i->touch, TOUCHES);
